litestar/contrib/pydantic.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from pydantic import BaseModel
+
+from litestar.dto.factory.abc import AbstractDTOFactory
+from litestar.dto.factory.field import DTO_FIELD_META_KEY, DTOField
+from litestar.dto.factory.types import FieldDefinition
+from litestar.dto.factory.utils import get_model_type_hints
+from litestar.types.empty import Empty
+from litestar.utils.helpers import get_fully_qualified_class_name
+
+if TYPE_CHECKING:
+    from typing import Any, ClassVar, Collection, Generator
+
+    from pydantic.fields import ModelField
+
+    from litestar.typing import ParsedType
+
+__all__ = ("PydanticDTO",)
+
+DataT = TypeVar("DataT", bound="BaseModel | Collection[BaseModel]")
+
+
+class PydanticDTO(AbstractDTOFactory[DataT], Generic[DataT]):
+    """Support for domain modelling with Pydantic."""
+
+    __slots__ = ()
+
+    model_type: ClassVar[type[BaseModel]]
+
+    @classmethod
+    def generate_field_definitions(cls, model_type: type[BaseModel]) -> Generator[FieldDefinition, None, None]:
+        for key, parsed_type in get_model_type_hints(model_type).items():
+            model_field = model_type.__fields__[key]
+
+            dto_field: DTOField | None = model_field.field_info.extra.get(DTO_FIELD_META_KEY)
+
+            def determine_default(_parsed_type: ParsedType, _model_field: ModelField) -> Any:
+                if _model_field.default_factory is not None:
+                    return Empty
+
+                if _model_field.default is None and not _parsed_type.is_optional:
+                    return Empty
+
+                return _model_field.default
+
+            field_def = FieldDefinition(
```

Comment:
Same issue as for msgspec, you need to copy info from `pydantic.FieldInfo`. See signature field stuff.